pyspectral/tests/test_rayleigh.py

```diff
@@ -188,19 +227,57 @@ def test_get_reflectance(self, get_effective_wvl,
         sun_zenith = np.array([67., 32.])
         sat_zenith = np.array([45., 18.])
         azidiff = np.array([150., 110.])
-        blueband = np.array([14., 5.])
+        redband_refl = np.array([14., 5.])
         retv = self.viirs_rayleigh.get_reflectance(
-            sun_zenith, sat_zenith, azidiff, 'M2', blueband)
+            sun_zenith, sat_zenith, azidiff, 'M2', redband_refl)
         self.assertTrue(np.allclose(retv, TEST_RAYLEIGH_RESULT1))
 
         sun_zenith = np.array([60., 20.])
         sat_zenith = np.array([49., 26.])
         azidiff = np.array([140., 130.])
-        blueband = np.array([12., 8.])
+        redband_refl = np.array([12., 8.])
         retv = self.viirs_rayleigh.get_reflectance(
-            sun_zenith, sat_zenith, azidiff, 'M2', blueband)
+            sun_zenith, sat_zenith, azidiff, 'M2', redband_refl)
         self.assertTrue(np.allclose(retv, TEST_RAYLEIGH_RESULT2))
 
+    @patch('os.path.exists')
+    @patch('pyspectral.utils.download_luts')
+    @patch('pyspectral.rayleigh.get_reflectance_lut_from_file')
+    @patch('pyspectral.rsr_reader.RelativeSpectralResponse.'
+           '_get_rsr_data_version')
+    @patch('pyspectral.rayleigh.Rayleigh.get_effective_wavelength')
+    def test_get_reflectance_dask_redband_outside_clip(self, get_effective_wvl,
+                                                       get_rsr_version, get_reflectance_lut_from_file,
+                                                       download_luts, exists):
+        """Test getting the reflectance correction with dask inputs - using red band reflections outside 0-100."""
+        rayl = da.from_array(TEST_RAYLEIGH_LUT, chunks=(10, 10, 10, 10))
+        wvl_coord = TEST_RAYLEIGH_WVL_COORD
+        azid_coord = da.from_array(TEST_RAYLEIGH_AZID_COORD, chunks=(1000,))
+        sunz_sec_coord = da.from_array(TEST_RAYLEIGH_SUNZ_COORD,
+                                       chunks=(1000,))
+        satz_sec_coord = da.from_array(TEST_RAYLEIGH_SATZ_COORD,
+                                       chunks=(1000,))
+
+        get_reflectance_lut_from_file.return_value = (rayl, wvl_coord, azid_coord,
+                                                      satz_sec_coord, sunz_sec_coord)
+        download_luts.return_code = None
+        exists.return_code = True
+        get_rsr_version.return_code = RSR_DATA_VERSION
+        get_effective_wvl.return_value = self.cwvl
+
+        sun_zenith = da.from_array(np.array([67., 32.]), chunks=2)
+        sat_zenith = da.from_array(np.array([45., 18.]), chunks=2)
+        azidiff = da.from_array(np.array([150., 110.]), chunks=2)
+        redband_refl = da.from_array(np.array([108., -0.5]), chunks=2)
+        retv = self.viirs_rayleigh.get_reflectance(sun_zenith, sat_zenith, azidiff, 'M2', redband_refl)
+        self.assertTrue(np.allclose(retv, TEST_RAYLEIGH_RESULT4))
+        self.assertIsInstance(retv, da.Array)
+
+        redband_refl = da.from_array(np.array([100., 20.0]), chunks=2)
```

Comment:
Should this be a separate test? It's cleaner when unit tests only test one thing

Reply:
Hmm, it is part of the same test I would say, namely that if you are outside the range 20-100 then all values give the same result...

Reply:
I it isbjust testing values outside thr valid range, then put all those values to the same call. If the point is to check the exact edges, then I'd say it is a separate test. So three test ranges: within valid range, outside valid range, the edge casea.